pkg/kubectl/cmd/set/helper.go

```diff
@@ -117,46 +117,41 @@ type Patch struct {
 	Patch  []byte
 }
 
+// CalculatePatch calls the mutation function on the provided info object, and generates a strategic merge patch for
+// the changes in the object. Encoder must be able to encode the info into the appropriate destination type.
+// This function returns whatever mutateFn returns.
+func CalculatePatch(patch *Patch, encoder runtime.Encoder, mutateFn func(*resource.Info) ([]byte, error)) bool {
+	patch.Before, patch.Err = runtime.Encode(encoder, patch.Info.Object)
+
+	patch.After, patch.Err = mutateFn(patch.Info)
+	if patch.Err != nil {
+		return true
+	}
+	if patch.After == nil {
+		return false
+	}
+
+	// TODO: should be via New
+	versioned, err := patch.Info.Mapping.ConvertToVersion(patch.Info.Object, patch.Info.Mapping.GroupVersionKind.GroupVersion())
+	if err != nil {
+		patch.Err = err
+		return true
+	}
+
+	patch.Patch, patch.Err = strategicpatch.CreateTwoWayMergePatch(patch.Before, patch.After, versioned)
+	return true
+}
+
 // CalculatePatches calls the mutation function on each provided info object, and generates a strategic merge patch for
 // the changes in the object. Encoder must be able to encode the info into the appropriate destination type. If mutateFn
 // returns false, the object is not included in the final list of patches.
-func CalculatePatches(infos []*resource.Info, encoder runtime.Encoder, mutateFn func(*resource.Info) (bool, error)) []*Patch {
+func CalculatePatches(infos []*resource.Info, encoder runtime.Encoder, mutateFn func(*resource.Info) ([]byte, error)) []*Patch {
```

Comment:
doc is now wrong.  update with what the `mutateFn` returns now.  Also, promote it to a real type and doc it there.